nf_core/modules/lint/__init__.py

```diff
@@ -148,121 +157,72 @@ def lint(self, module=None, all_modules=False, print_results=True, show_passed=F
         if module:
             log.info(f"Linting module: [magenta]{module}")
 
-        # Check that supplied test keys exist
-        bad_keys = [k for k in self.key if k not in self.lint_tests]
-        if len(bad_keys) > 0:
-            raise AssertionError(
-                "Test name{} not recognised: '{}'".format(
-                    "s" if len(bad_keys) > 1 else "",
-                    "', '".join(bad_keys),
-                )
-            )
-
-        # If -k supplied, only run these tests
-        if self.key:
-            log.info("Only running tests: '{}'".format("', '".join(self.key)))
-            self.lint_tests = [k for k in self.lint_tests if k in self.key]
+        # Filter the tests by the key if one is supplied
+        if key:
+            self.filter_tests_by_key()
```

Comment:
Don't we need to supply the `key` to the filter function here?

Reply:
You're right. Went back and forth between having `key` as a argument and as an attribute of the class.